ProcessLib/HT/HTFEM.h

```diff
@@ -88,6 +88,76 @@ class HTFEM : public HTLocalAssemblerInterface
         return Eigen::Map<const Eigen::RowVectorXd>(N.data(), N.size());
     }
 
+    /// Computes the flux in the point \c pnt_local_coords that is given in
+    /// local coordinates using the values from \c local_x.
+    // TODO add time dependency
+    std::vector<double> getFlux(
+        MathLib::Point3d const& pnt_local_coords,
+        std::vector<double> const& local_x) const override
+    {
+        // eval dNdx and invJ at given point
+        using FemType =
+            NumLib::TemplateIsoparametric<ShapeFunction, ShapeMatricesType>;
+
+        FemType fe(*static_cast<const typename ShapeFunction::MeshElement*>(
+            &this->_element));
+
+        typename ShapeMatricesType::ShapeMatrices shape_matrices(
+            ShapeFunction::DIM, GlobalDim, ShapeFunction::NPOINTS);
+
+        // Note: Axial symmetry is set to false here, because we only need dNdx
+        // here, which is not affected by axial symmetry.
+        fe.computeShapeFunctions(pnt_local_coords.getCoords(), shape_matrices,
+                                 GlobalDim, false);
+        std::vector<double> flux;
+        flux.resize(3);
+
+        // fetch permeability, viscosity, density
+        SpatialPosition pos;
+        pos.setElementID(this->_element.getID());
+
+        MaterialLib::Fluid::FluidProperty::ArrayType vars;
+        double T_int_pt = 0.0;
+        double p_int_pt = 0.0;
+
+        // local_x contains the local temperature and pressure values
+        NumLib::shapeFunctionInterpolate(local_x, shape_matrices.N, T_int_pt, p_int_pt);
+
+        // set the values into array vars
+        vars[static_cast<int>(MaterialLib::Fluid::PropertyVariableType::T)] =
```

Comment:
Would it make sense to pass the vars[T] directly to shapeFunctionInterpolate call?
```c++
NumLib::shapeFunctionInterpolate(local_x, shape_matrices.N, vars[static_cast<int>(MaterialLib::Fluid::PropertyVariableType::T)], vars[static_cast<int>(MaterialLib::Fluid::PropertyVariableType::p)]);
```